save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/TopBar.kt

```diff
@@ -319,5 +233,83 @@ fun topBar() = FC<TopBarProps> { props ->
     }
 }
 
-private fun textColor(hrefAnchor: String, location: history.Location) =
+/**
+ * [Props] of the top bor component
+ */
+external interface TopBarProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+
+    /**
+     * true if the device is mobile (screen is less 1000px)
+     */
+    var isMobileScreen: Boolean?
+}
+
+/**
+ * [Props] of the top bor component with [location]
+ */
+external interface TopBarPropsWithLocation : TopBarProps {
+    /**
+     * Currently logged in user or null
+     */
+    var location: Location
+}
+
+/**
+ * @property hrefAnchor is link
+ * @property jsoWidth is width of the link text
+ * @property text is link text
+ */
+data class TopBarLink(
+    val hrefAnchor: String,
+    val jsoWidth: Width,
+    val text: String,
+)
+
+private fun ChildrenBuilder.dropdownEntry(
+    faIcon: dynamic,
+    text: String,
+    handler: ChildrenBuilder.(ButtonHTMLAttributes<HTMLButtonElement>) -> Unit = { },
+) = button {
+    type = ButtonType.button
+    className = ClassName("btn btn-no-outline dropdown-item rounded-0 shadow-none")
+    fontAwesomeIcon(icon = faIcon) {
+        it.className = "fas fa-sm fa-fw mr-2 text-gray-400"
+    }
+    +text
+    handler(this)
+}
+
+/**
+ * A component for web page top bar
+ *
+ * @return a function component
+ */
+fun topBar() = FC<TopBarProps> { props ->
+    val thisLocation = useLocation()
+    nav {
+        className =
+                ClassName("navbar navbar-expand navbar-dark bg-dark topbar mb-3 static-top shadow mr-1 ml-1 rounded")
+        id = "navigation-top-bar"
+        topBarUrlSplits {
+            userInfo = props.userInfo
+            isMobileScreen = props.isMobileScreen
+            location = thisLocation
+        }
+        topBarLinks {
+            userInfo = props.userInfo
```

Comment:
We don't use `userInfo` in this component, do we?

Reply:
Deleted this